python/dgl/sampling/labor.py

```diff
@@ -0,0 +1,276 @@
+#
+#   Copyright (c) 2022 by Contributors
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+#   Based off of neighbor.py
+#
+
+
+"""Labor sampling APIs"""
+
+from .._ffi.function import _init_api
+from .. import backend as F
+from ..base import DGLError, NID, EID
+from ..heterograph import DGLHeteroGraph
+from .. import ndarray as nd
+from .. import utils
+from .utils import EidExcluder
+from ..backend import from_dgl_nd
+
+__all__ = [
+    'sample_labors']
+
+def sample_labors(g, nodes, fanout, random_seed, edge_dir='in', prob=None,
+                     importance_sampling=0,
+                     copy_ndata=True, copy_edata=True, exclude_edges=None, output_device=None):
+    """Sample neighboring edges of the given nodes and return the induced subgraph.
+
+    For each node, a number of inbound (or outbound when ``edge_dir == 'out'``) edges
+    will be randomly chosen.  The graph returned will then contain all the nodes in the
+    original graph, but only the sampled edges.
+
+    Node/edge features are not preserved. The original IDs of
+    the sampled edges are stored as the `dgl.EID` feature in the returned graph.
+
+    Parameters
+    ----------
+    g : DGLGraph
+        The graph.  Can be either on CPU or GPU.
```

Comment:
Does it work for heterogeneous graph?  Better document it if so.

Reply:
It does, just like neighbor sampler